### PR TITLE
Allow to use config_retrieve on whole directory

### PR DESCRIPTION
**Pull request by BannZay**
Work on topmost directory only

### CODE REVIEW

File: src/tools/config_common.h [hunk 1]
```diff
@@ -10,28 +53,15 @@ int main(int argc, const char **argv)
 		dbg_msg("usage", "%s FILE1 [ FILE2... ]", argv[0]);
```
Comment (heinrich5991) [inline src/tools/config_common.h:53]:
```suggestion
		dbg_msg("usage", "%s FILE1 [ FILE2... ]", argv[0]);
		dbg_msg("usage", "%s DIRECTORY", argv[0]);
```

Please add this usage to the possible forms, too.

File: src/tools/config_common.h [hunk 2]
```diff
@@ -1,6 +1,49 @@
...
+	Process(pStorage, pItemName, aConfig);
+}
+
+static int ListdirCallback(const char *pItemName, int IsDir, int StorageType, void *pUser)
+{
+	if(!IsDir)
+	{
+		ListDirectoryContext Context = *((ListDirectoryContext *)pUser);
+		char aName[2048];
+		str_format(aName, sizeof(aName), "%s\\%s", Context.pPath, pItemName);
```
Comment (heinrich5991) [inline src/tools/config_common.h:40]:
```suggestion
		str_format(aName, sizeof(aName), "%s/%s", Context.pPath, pItemName);
```

Use slash, backslash only works on Windows.

**Review by heinrich5991 — COMMENTED** [review]
Thanks for explaining your use case on Discord.

> [11:05 PM] heinrich5991: @BannZay what is the use case for that pull request?
> [11:06 PM] heinrich5991: it seems to me the same can be done on windows by dragging all the map files onto the executable, or on linux by executing something similar to ./config_retrieve maps/*.map
> […]
> [11:24 PM] BannZay: @heinrich5991 it is possible until the amount of the files is not that big. I tried to drag n drop all DDNet maps (which is 1700+) but it didnt work

Perhaps you could include that reason in the commit description, so future developers don't wonder what this is for. :)

Looks good to me, thanks for sharing your code. Please fix the review comments.

**Comment by BannZay** [comment]
> Perhaps you could include that reason in the commit description

The reason I didn't is because it is nice to have such possibility regardless of limitations of others approaches

**Comment by heinrich5991** [comment]
bors r+

Thanks.

**Pull request merged**